service/gateway/object_handler.go

```diff
@@ -356,11 +360,41 @@ func (gateway *Gateway) getObjectByUniversalEndpointHandler(w http.ResponseWrite
 		return
 	}
 
-	// In first phase, do not provide universal endpoint for private object
+	// check when trying to access private object. Object name, bucket name, expire time, signature must all match
+	if reqContext.expireTime != "" && reqContext.signature != "" {
+		if expires, err = time.Parse(time.RFC3339, reqContext.expireTime); err != nil {
+			log.Errorw("failed to parse expire time", "error", err)
+			errDescription = makeErrorDescription(err)
+			return
+		}
+		if expires.Before(time.Now()) {
```

Comment:
I feel the limit shall be added at the time of generating the signature. Otherwise the signature would be successfully generated but refused for use, making users confused.